libs/atqr/domain/src/lib/contact/contact.valueObject.ts

```diff
@@ -0,0 +1,38 @@
+import { MaxLength, MinLength } from 'class-validator';
+import { EmailAddress } from '../EmailService';
+import { ValueObject } from '../../utils/valueObject';
+
+export enum ParticipationRole {
+  Contender,
+  Judge,
+}
+export class Contact extends ValueObject<Contact> {
+  @MinLength(5)
+  @MaxLength(20)
+  name: string;
+  email: EmailAddress;
+  role: ParticipationRole;
+
+  constructor(name: string, email: EmailAddress, role: ParticipationRole) {
```

Comment:
Use `public readonly prop: type` and you can remove property declarations

Reply:
@svallory I did it because decorators can not be used inside constructor param . Looks better with all proprieties declared than only name and its decorators declared and the other two just public readonly inside of the constructor params , right ?